pkg/compiler/codegen.go

```diff
@@ -1158,7 +1159,8 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 				c.convertStruct(lit, true)
 				return nil
 			}
-			c.prog.Err = fmt.Errorf("'&' can be used only with struct literals")
+			pos := c.buildInfo.config.Fset.Position(n.Pos())
+			c.prog.Err = fmt.Errorf("%s: '&' can be used only with struct literals", pos)
```

Comment:
There's a couple of places where the same pattern may be applied (e.g. 20 lines below, try to search for `c.prog.Err =` usages). Let's move it to a helper and apply it to all relevant places.